pkg/controller/nodeipam/ipam/range_allocator.go

```diff
@@ -284,6 +291,13 @@ func (r *rangeAllocator) ReleaseCIDR(node *v1.Node) error {
 			return fmt.Errorf("failed to parse CIDR %s on Node %v: %v", cidr, node.Name, err)
 		}
 
+		// if node has a pre allocate cidr that does not exist in our cidrs
+		// this will happen if cluster went from dualstack(multi cidrs) to non-dualstack
+		// then we have now way of locking it
+		if idx >= len(r.cidrSets) {
+			return fmt.Errorf("node:%s has an allocated cidr: %v at index:%v that does not exist in cluster cidrs configuration. Update node.cidrs by removing this cidr then restart controller manager", node.Name, cidr, idx)
```

Comment:
Don't you have to delete the node in this case, not update `--cluster-cidrs` in CM?

Reply:
the reference is `Node.Spec.CIDRs` not `--cluster-cidrs`, will modify